### DISCUSSION TITLE
Bump go version to at least 1.18?

### DISCUSSION THREAD

**Discussion by rszyma** [pelletier/go-toml#938] [General]
I'd like to propose a PR that utilizes generics, but generics are only supported since `1.18`, while this lib is `1.16`.

Is there any reason to keep at `1.16`?

**Comment by pelletier** [comment]
My only hard rule is to support at least [the last two Go versions](https://github.com/pelletier/go-toml?tab=readme-ov-file#versioning). In practice, I avoid breaking compatibility without a good reason. I am open to bumping the minimum required Go version to 1.18 to start using generics, but I'd like to see examples of how they improve the library.
